### PR TITLE
E2E: Flow System (happy paths) + Release (patch): 0.204.4

### PR DESCRIPTION
**Pull request by s373r**
### Fixed
- Added several E2E tests (happy paths) covering the Flows tab in the UI
- Corrected behavior of `MySqlAccountRepository::get_accounts_by_ids()`, for the case of empty IDs collection

## Checklist before requesting a review

- [x] Unit and integration tests added ✅
  
- [x] Compatibility:
    
  - [x] Network APIs: ✅
    
  - [x] Workspace layout and metadata: ✅
    
  - [x] Configuration: ✅
    
  - [x] Container images: ✅
- [x] Observability: 
    
  - [x] Tracing / [metrics](https://github.com/kamu-data/kamu-standards/blob/master/metrics_design.md): ❌ not needed
    
  - [x] Alerts: ❌ not needed
- [x] Documentation:
    
  - [x] [Changelog](./CHANGELOG.md): ✅
    
  - [x] [Public documentation](https://github.com/kamu-data/kamu-docs/): ❌ not needed
  
- [x] Downstream effects:
    
  - [x] [kamu-node](https://github.com/kamu-data/kamu-node): ❌ not needed
    
  - [x] [kamu-web-ui](https://github.com/kamu-data/kamu-web-ui): ❌ not needed
    
  - [x] [release-queue](https://github.com/orgs/kamu-data/projects/4/views/1) ❌ not needed

### CODE REVIEW

File: src/app/cli/src/cli_commands.rs [hunk 1]
```diff
@@ -263,7 +263,7 @@ pub fn get_command(
                     work_catalog.get_one()?,
                     work_catalog.get_one()?,
                     c.dataset.unwrap_or_default(),
-                    work_catalog.get_one()?,
+                    cli_catalog.get_one()?,
```
Comment (s373r) [inline src/app/cli/src/cli_commands.rs:266]: `current_account_subject`

File: src/app/cli/src/cli_commands.rs [hunk 2]
```diff
@@ -279,7 +279,7 @@ pub fn get_command(
             work_catalog.get_one()?,
             work_catalog.get_one()?,
             c.dataset.unwrap_or_default(),
-            work_catalog.get_one()?,
+            cli_catalog.get_one()?,
```
Comment (s373r) [inline src/app/cli/src/cli_commands.rs:282]: `current_account_subject`

File: src/app/cli/src/cli_commands.rs [hunk 3]
```diff
@@ -407,13 +406,13 @@ pub fn get_command(
                         sc.external_address,
                         sc.get_token,
                         work_catalog.get_one()?,
-                        work_catalog.get_one()?,
+                        cli_catalog.get_one()?,
```
Comment (s373r) [inline src/app/cli/src/cli_commands.rs:409]: `current_account_subject`

**Review by s373r — COMMENTED** [review]
Add notes to readers

File: src/app/cli/src/app.rs [hunk 4]
```diff
@@ -197,9 +197,7 @@ pub async fn run(workspace_layout: WorkspaceLayout, args: cli::Cli) -> Result<()
             .build();
         let maybe_server_catalog = if cli_commands::command_needs_server_components(&args) {
             let server_catalog =
-                configure_server_catalog(&final_base_catalog, is_multi_tenant_workspace)
-                    .add_value(current_account.to_current_account_subject())
-                    .build();
+                configure_server_catalog(&final_base_catalog, is_multi_tenant_workspace).build();
```
Comment (zaychenko-sergei) [inline src/app/cli/src/app.rs:200]: There should not be a need in "multi-tenancy" marker for server catalog

File: src/app/cli/src/app.rs [hunk 5]
```diff
@@ -243,15 +236,24 @@ pub async fn run(workspace_layout: WorkspaceLayout, args: cli::Cli) -> Result<()
...
+            error = %e.pretty(true),
+            "Initialize base components failed",
+        );
+        return Err(e);
+    }
+
     let is_transactional =
         maybe_db_connection_settings.is_some() && cli_commands::command_needs_transaction(&args);
-    let work_catalog = maybe_server_catalog.unwrap_or(cli_catalog);
+    let work_catalog = maybe_server_catalog.unwrap_or(cli_catalog.clone());
```
Comment (zaychenko-sergei) [inline src/app/cli/src/app.rs:250]: The transaction should happen on CLI catalog always, as this is what commands inject.
If server is needed, it should be in the chain of CLI catalog.

File: src/app/cli/src/commands/system_api_server_gql_query_command.rs [hunk 6]
```diff
@@ -12,15 +12,15 @@ use dill::Catalog;
 use super::{CLIError, Command};
 
 pub struct APIServerGqlQueryCommand {
-    base_catalog: Catalog,
+    server_catalog: Catalog,
```
Comment (zaychenko-sergei) [inline src/app/cli/src/commands/system_api_server_gql_query_command.rs:15]: This clarification is a nice idea!

File: src/app/cli/src/explore/api_server.rs [hunk 7]
```diff
@@ -51,11 +50,11 @@ impl APIServer {
     ) -> Result<Self, InternalError> {
         // Background task executor must run with server privileges to execute tasks on
         // behalf of the system, as they are automatically scheduled
-        let task_executor = cli_catalog.get_one().unwrap();
+        let task_executor = server_catalog.get_one().unwrap();
```
Comment (zaychenko-sergei) [inline src/app/cli/src/explore/api_server.rs:53]: Don't understand this change

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
Stabilized and commited my smaller hotfix.
Plz take those correcitons, and let's finalize the greate test ideas introduced here.

File: src/app/cli/src/app.rs (hunk 4 above)
Comment (s373r) [reply src/app/cli/src/app.rs]: Removed (by rebasing on your changes) -- Done

File: src/app/cli/src/app.rs (hunk 5 above)
Comment (s373r) [reply src/app/cli/src/app.rs]: I get the idea.

Done (by rebasing on your changes)

File: src/app/cli/src/commands/system_api_server_gql_query_command.rs (hunk 6 above)
Comment (s373r) [reply src/app/cli/src/commands/system_api_server_gql_query_command.rs:15]: I am also thinking about typed catalog (via new type idiom)

File: src/app/cli/src/explore/api_server.rs (hunk 7 above)
Comment (s373r) [reply src/app/cli/src/explore/api_server.rs]: Now I see what you mean -- rolled it back

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**